include/kernel.h

```diff
@@ -2476,8 +2477,13 @@ struct k_sem {
  *
  * @return N/A
  */
-extern void k_sem_init(struct k_sem *sem, unsigned int initial_count,
-			unsigned int limit);
+static inline void k_sem_init(struct k_sem *sem, unsigned int initial_count,
+			      unsigned int limit);
+
+K_SYSCALL_DECLARE3_VOID(K_SYSCALL_SEM_INIT, k_sem_init,
```

Comment:
Since this macro is only for the Python script to generate code, do we need to name parameters? Prototypes can have only types.